src/routing/comm.rs

```diff
@@ -30,23 +32,30 @@ pub(crate) struct Comm {
 }
 
 impl Comm {
-    pub fn new(
+    pub async fn new(
         transport_config: qp2p::Config,
         event_tx: mpsc::Sender<ConnectionEvent>,
     ) -> Result<Self> {
         let quic_p2p = QuicP2p::with_config(Some(transport_config), Default::default(), true)?;
 
         // Don't bootstrap, just create an endpoint where to listen to
         // the incoming messages from other nodes.
-        let endpoint = quic_p2p.new_endpoint()?;
+        let (endpoint, incoming_connections, incoming_messages, disconnections) =
+            quic_p2p.new_endpoint().await?;
 
-        let _ = task::spawn(handle_incoming_connections(
-            endpoint.listen(),
+        let _ = task::spawn(handle_incoming_messages(
+            incoming_messages,
+            event_tx.clone(),
+        ));
+
+        let _ = task::spawn(handle_disconnection_events(
+            disconnections,
             event_tx.clone(),
         ));
 
         Ok(Self {
             _quic_p2p: quic_p2p,
+            _incoming_connections: incoming_connections,
```

Comment:
Wouldn't this leak memory if we are not reading from this stream?

Reply:
So, could we simply drop this stream if we are not going to use it, or would that cause problems?

Reply:
Yes we can do that. We'll need to ignore the result of `tx.send()` in qp2p but I guess that's okay?

Reply:
Yep, sounds good.